src/transport.js

```diff
@@ -53,11 +54,18 @@ module.exports = function (swarm) {
       // specific for the transport we are using
       const proxyConn = new Connection()
 
-      next(multiaddrs.shift())
+      each(multiaddrs, next)
```

Comment:
Cool, I'll leave it up to you to make this change as I'm not well versed in the async functions. I'm glad I could bring the issue to the table :)